llvm/lib/IR/Type.cpp

```diff
@@ -792,28 +793,50 @@ TargetExtType::TargetExtType(LLVMContext &C, StringRef Name,
 TargetExtType *TargetExtType::get(LLVMContext &C, StringRef Name,
                                   ArrayRef<Type *> Types,
                                   ArrayRef<unsigned> Ints) {
+  return cantFail(getOrError(C, Name, Types, Ints));
```

Comment:
I think this will still call `checkParams` even in a Release build. It would be nice to avoid that, but to do that I think I would need to duplicate most of the body of `getOrError` here.

Reply:
I think it's ok as it only happens the first time the type is created, so the cost is probably negligible.